sdk/src/main/java/com/bugsnag/android/Error.java

```diff
@@ -18,17 +19,14 @@
  */
 public class Error implements JsonStream.Streamable {
 
-    @SuppressWarnings("NullableProblems") // set after construction
     @NonNull
-    private Map<String, Object> appData;
+    private Map<String, Object> appData = new HashMap<String, Object>();
 
-    @SuppressWarnings("NullableProblems") // set after construction
     @NonNull
-    private Map<String, Object> deviceData;
+    private Map<String, Object> deviceData = new HashMap<String, Object>();
 
-    @SuppressWarnings("NullableProblems") // set after construction
     @NonNull
-    private User user;
+    private User user = new User();
```

Comment:
The fact that they are marked as `@NonNull` but actually could be null led to situations where there would be a native event but these fields weren't initialized, leading to a crash. It took a while to debug because they were also suppressing the nullability warning.